include/re_tls.h

```diff
@@ -30,8 +30,8 @@ enum tls_keytype {
 };
 
 
-int tls_alloc(struct tls **tlsp, enum tls_method method, const char *keyfile,
-	      const char *pwd);
+int tls_alloc(struct tls **tlsp, enum tls_method method, uint16_t timeout,
+	      const char *keyfile, const char *pwd);
```

Comment:
this change breaks the tls api ...

Reply:
We could instead add a setter for the timeout.